torch/fx/experimental/fx2trt/example/sparse_quantized_resnet_test.py

```diff
@@ -0,0 +1,316 @@
+        sparse_params = get_sparse_params(sparsifier)
+        logger.log(logger.INFO, f"Sparse params: {sparse_params}")
+
+    return model
+
+
+def build_fp16_trt(rn18, device=None, logger_level=None):
+    rn18 = copy.deepcopy(rn18)
+    rn18 = acc_tracer.trace(rn18, [torch.randn(1, 3, 224, 224)])
+    interp = TRTInterpreter(
+        rn18, [InputTensorSpec(torch.Size([3, 224, 224]), torch.float, has_batch_dim=False)],
+        logger_level=logger_level)
+    interpreter_result = interp.run(fp16_mode=True)
+    return TRTModule(interpreter_result.engine, interpreter_result.input_names, interpreter_result.output_names)
+
+
+@torch.no_grad()
+def build_sparse_fp16_trt(rn18, sparse_params=None, logger_level=None):
+    r"""Builds a sparse model based on the rn18.
+
+    Args:
+        sparse_params: Sparsity parameters to pass to the sparsifier.
+
+    Note: Once the model is sparsified, it stores the sparse tensors as dense
+          with whole bunch of zeros. To see the benefits of sparsity, use
+          the A100 sparse kernels with the following sparse parameters:
+          - sparsity_level=1.0
+          - sparse_block_shape=(1, 4)
+          - zeros_per_block=2
+          When using on non-specialized CUDA kernels such as cuSparse, there
+          will be no latency benefits.
+    """
+    rn18 = copy.deepcopy(rn18)
+
+    # Sparsify the model
+    post_training_sparsify(rn18, verbose=True)
+
+    # Fx the model
+    rn18 = acc_tracer.trace(rn18, [torch.randn(1, 3, 224, 224)])
+    interp = TRTInterpreter(
+        rn18, [InputTensorSpec(torch.Size([3, 224, 224]), torch.float, has_batch_dim=False)],
+        logger_level=logger_level)
+    interpreter_result = interp.run(fp16_mode=True, sparse_weight=True)
+    return TRTModule(interpreter_result.engine, interpreter_result.input_names, interpreter_result.output_names)
+
+
+@torch.no_grad()
+def build_int8_trt(rn18, logger_level=None):
+    rn18 = copy.deepcopy(rn18)
+    data = torch.randn(1, 3, 224, 224)
+    # data = torch.randn(1, 32)
+    # data = torch.randn(1, 64, 10, 10)
+    # TensorRT only supports symmetric quantization
+    qconfig = torch.ao.quantization.QConfig(
+        activation=torch.ao.quantization.observer.HistogramObserver.with_args(
+            qscheme=torch.per_tensor_symmetric, dtype=torch.qint8
+        ),
+        # weight=torch.ao.quantization.default_weight_observer
+        # uncomment to check per channel quant works
+        weight=torch.quantization.default_per_channel_weight_observer
+    )
+    prepared = prepare_fx(rn18, {"": qconfig})
+    for _ in range(10):
+        prepared(data)
+    quantized_rn18 = convert_fx(prepared, is_reference=True)
+    ref_res = quantized_rn18(data)
+    print("quantized model:", quantized_rn18)
+
+    quantized_rn18 = acc_tracer.trace(quantized_rn18, [data])  # type: ignore[assignment]
+    interp = TRTInterpreter(
+        quantized_rn18,
+        [InputTensorSpec(torch.Size([-1, *data.shape[1:]]), torch.float,
```

Comment:
We recently added a new api to generate dynamic input tensor spec in case you're interested https://github.com/pytorch/pytorch/blob/05946051f89cb3b5863dee6d3e9fb93dd49fd63e/torch/fx/experimental/fx2trt/input_tensor_spec.py#L67.